pkg/routing/routing_linux.go

```diff
@@ -164,6 +173,75 @@ func GetRoute(ctx context.Context, routedNet *net.IPNet) (*Route, error) {
 	}, nil
 }
 
+func openTable(ctx context.Context) (Table, error) {
+	rules, err := netlink.RuleList(netlink.FAMILY_ALL)
+	if err != nil {
+		return nil, err
+	}
+	// Sort the rules by index ascending to make sure we find an open one
+	sort.Slice(rules, func(i, j int) bool {
+		return rules[i].Table < rules[j].Table
+	})
+	index := 775
+	priority := 32766 // default initial priority
+	for _, rule := range rules {
+		dlog.Tracef(ctx, "Found routing rule %+v", rule)
+		if rule.Table == 0 || rule.Table == 255 {
+			// System rules, ignore
+			continue
+		}
+		if rule.Priority <= priority {
+			priority = rule.Priority - 1
+		}
+		if rule.Table == index {
+			// There's already a table with the default index, get a new one
+			index++
+		}
+	}
+	dlog.Infof(ctx, "Creating routing table with index %d and priority %d", index, priority)
+	rule := netlink.NewRule()
+	rule.Table = index
+	rule.Priority = priority
+	rule.Family = netlink.FAMILY_V4
+	if err := netlink.RuleAdd(rule); err != nil {
+		return nil, err
+	}
+	return &table{
+		index: index,
+		rule:  rule,
+	}, nil
+}
```

Comment:
Still curious, why do we want the priority to be under the existing rules ?

Reply:
@knlambert Because we don't want to magically change the existing rules without approval from the user.

Reply:
well having the priority _under_ the existing rules will mean our table gets queried _first_
we do this so that overrides actually work; we do the check for existing routes elsewhere